projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php

```diff
@@ -907,6 +915,20 @@ function wpcom_launchpad_is_paid_newsletter_enabled() {
 	return wpcom_launchpad_has_goal_paid_subscribers() && apply_filters( 'wpcom_launchpad_intent_paid_newsletter_enabled', false );
 }
 
+/**
+ * Checks if site has newsletter or write as site intent.
+ *
+ * @return bool True if the task list is enabled, false otherwise.
+ */
+function wpcom_launchpad_is_newsletter_or_write_intent() {
+	$intent = get_option( 'site_intent', false );
+	if ( 'newsletter' === $intent || 'write' === $intent ) {
```

Comment:
Can we restrict this a little more? I tested with a Free Newsletter and a site with the Write intent, and both returned `enabled = true`, but the checklist was empty. So maybe it shouldn't be enabled when the checklist is empty.

<img width="1149" alt="Screen Shot 2023-09-28 at 17 13 24" src="https://github.com/Automattic/jetpack/assets/1234758/b617064d-1619-46f4-a58f-962464d168f5">

Reply:
> but the checklist was empty

Hmm. Not able to duplicate this. I wouldn't think an empty checklist would be related to the is_enabled callback? If you're seeing an empty checklist, that would seem like it would reflect some other other issue wouldn't it? 

To put it otherwise, if I send a request to `/sites/YOURDOMAIN/launchpad?checklist_slug=earn-newsletter`, shouldn't I **always** get back the earn-newsletter checklist with two items?

Reply:
To confirm, you'll get 0 checklist items if the site_goals don't include paid_subscribers because the task definitions for set_up_payments and create_paid_newsletter both include `'is_visible_callback' => 'wpcom_launchpad_has_goal_paid_subscribers',`

I need to think how to handle that. I need to be able to load those tasks even if a site doesn't have that goal. 

Options: 
   - Just set a site_goal of paid_newsletters when someone visits the earn page (since they've expressed monetization intent)
   - Create different tasks that don't depend on that is_visible callback.

Reply:
FYI - I recall now that is_visible callback was originally set because we're using the same task definitions for free and paid newsletters and didn't want to show them on paid newsletters. But that intention doesn't travel well at this point, later, when we want to encourage those with free newsletters to monetize. 

It's also a bit odd / non-ideal because I've created a checklist specifically to fetch those two items, but I can't do so without updating the site_goals.

Reply:
I've made several changes based on this and also broader discussion about Launchpad on Earn:
   - I've created two new tasks for for stripe_connected and paid_offer_created that don't have the limitation of needing to have a paid_newsletter goal. 
   - Both tasks above have __return_true as their is_visible callback. For now, I can't think of any reason we would not want those visible on the Earn launchpad.
   - I've updated the checklist slug to be 'earn' instead of 'earn-newsletter'. We had been planning to add a general launchpad to earn after doing it for newsletters. I'm reversing the order and doing the general launchpad first.

Reply:
Just a note that the method for which this comment was added has been removed because I've updated this PR to be generic, and to show the earn launchpad for all sites, not just newsletter / blog / write.